webapp/redux/middleware/middleware.js

```diff
@@ -106,6 +106,33 @@ const simulateNetwork = (payload) => createSimulateBackendCall(
   GEPPETTO.Resources.RUNNING_SIMULATION,
 );
 
+class ErrorDialogManager {
+  errorTag = 'SimulationId ';
+  errorIds = [];
```

Comment:
The use of an array data structure as a lookup table is not ideal. A set or a map are way more efficient for lookup operations